key/src/ed25519/private.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::Public;
+use primitives::H512;
+use sodiumoxide::crypto::sign::{gen_keypair, SecretKey, SECRETKEYBYTES};
+use std::ops::Deref;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Private(SecretKey);
+
+impl Private {
+    pub fn random() -> Self {
+        let (_, secret) = gen_keypair();
+        Private(secret)
+    }
+
+    pub fn public_key(&self) -> Public {
+        self.0.public_key().into()
+    }
+
+    pub fn from_slice(slice: &[u8]) -> Option<Self> {
+        SecretKey::from_slice(slice).map(Self)
+    }
+}
+
+impl FromStr for Private {
+    type Err = crate::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let for_slice = H512::from_str(s).map_err(|e| crate::Error::Custom(format!("{:?}", e)))?;
+        Ok(SecretKey::from_slice(&for_slice).expect("H512 has length 64").into())
+    }
+}
+
+impl Default for Private {
```

Comment:
The only use case of `Default` of `Private` is `Drop` of `DecrptedAccount`, but it can be removed because `SecretKey` does the role of `Drop` of `DecrptedAccount`.

Reply:
Ok that's great